### PR TITLE
Add new structure for permissions API

### PR DESCRIPTION
**Pull request by nshuba**
**Asana Task/Github Issue:** https://app.asana.com/0/0/1205985283194932/f

## Description
* Match Chrome on what names are returned by the Permissions API on Android
* Pass query for camera and mic to the native platform on Android

**Steps to test this PR**:
_Requirements_
Get the latest Android app and apply config changes from this PR: you can use https://jsonblob.com/api/jsonBlob/1160261143960084480 which already includes them

_Old app, new config_
- [ ] Go to https://nshuba.github.io/test-site/ and select "camera" from the drop-down
- [ ] Use the dropdown to verify camera and mic permissions always return 'prompt' - we're just testing backwards compatibility here

_Updated app, new config_
- [ ] Follow testing steps in https://github.com/duckduckgo/content-scope-scripts/pull/861

#### Reference
- [Config Reviewer Documentation](https://app.asana.com/0/1200890834746050/1204443212791216/f)
- [Config Maintainer Documentation](https://app.asana.com/0/1200890834746050/1200573250322769/f)
- [Feature Implementer Documentation](https://app.asana.com/0/1200890834746050/1201498956177210/f)

### CODE REVIEW

File: overrides/android-override.json [hunk 1]
```diff
@@ -393,9 +393,7 @@
                             {
                                 "op": "add",
                                 "path": "/cleanIframeValue",
-                                "value": {
```
[the remark this answers is not in the document]
Comment (jonathanKingston) [reply overrides/android-override.json:396]: Both are valid and the same. This is the code: https://github.com/duckduckgo/content-scope-scripts/blob/88feb44ea7d090981730b0105720c9f905804ddc/src/content-feature.js#L191-L195

I think we can revert this and the viewport change.

[the remark this answers is not in the document]
Comment (jonathanKingston) [reply overrides/android-override.json:396]: I think the schema definition for domains is overly simplistic to what we needed at the time.


"value" can be any type, it's expected to match the feature itself.

Reading the validation I don't know why it's not currently working? Were we not running it before?

[the remark this answers is not in the document]
Comment (nshuba) [reply overrides/android-override.json:396]: That's right, we were not running the validation before. If both values are acceptable maybe it's ok to keep so validation can pass?

[the remark this answers is not in the document]
Comment (jonathanKingston) [reply overrides/android-override.json:396]: I double checked the code in C-S-S and it seems fine to change.

Really I was hoping to move away from this string syntax for enabling features.

This patching code can actually take any JSON permitted value for example the following is supported:

```json
                    {
                        "domain": "test.com",
                        "patchSettings": [
                            {
                                "op": "add",
                                "path": "/permissions/supportedPermissions/push",
                                "value":  {
                                    "name": "thing",
                                    "native": true
                                }
                            }
                        ]
                    },
```

IMO the correct fix is to change https://github.com/duckduckgo/privacy-configuration/blob/main/tests/schemas/webcompat-settings.json

```json
                                "value": {
                                    "type": "string",
                                    "description": "The value to replace at the specified path"
                                }
```

becomes
```json
                                "value": {
                                     "oneOf": [
                                        {
                                            "type": "string"
                                        },
                                        {
                                            "type": "array"
                                        },
                                        {
                                            "type": "object"
                                        },
                                        {
                                            "type": "number"
                                        }
                                      ],
                                    "description": "The value to replace at the specified path"
                                }
```

[the remark this answers is not in the document]
Comment (nshuba) [reply overrides/android-override.json]: Ah that makes sense - thank you! Updated :)

**Review by jonathanKingston — APPROVED** [review]

**Pull request merged**